gno.land/pkg/sdk/vm/handler.go

```diff
@@ -84,27 +81,58 @@ const (
 	QueryFile    = "qfile"
 )
 
-func (vh vmHandler) Query(ctx sdk.Context, req abci.RequestQuery) (res abci.ResponseQuery) {
-	switch secondPart(req.Path) {
+func (vh vmHandler) Query(ctx sdk.Context, req abci.RequestQuery) abci.ResponseQuery {
+	var (
+		res  abci.ResponseQuery
+		path = secondPart(req.Path)
+	)
+
+	switch path {
 	case QueryPackage:
-		return vh.queryPackage(ctx, req)
+		res = vh.queryPackage(ctx, req)
 	case QueryStore:
-		return vh.queryStore(ctx, req)
+		res = vh.queryStore(ctx, req)
 	case QueryRender:
-		return vh.queryRender(ctx, req)
+		res = vh.queryRender(ctx, req)
 	case QueryFuncs:
-		return vh.queryFuncs(ctx, req)
+		res = vh.queryFuncs(ctx, req)
 	case QueryEval:
-		return vh.queryEval(ctx, req)
+		res = vh.queryEval(ctx, req)
 	case QueryFile:
-		return vh.queryFile(ctx, req)
+		res = vh.queryFile(ctx, req)
 	default:
-		res = sdk.ABCIResponseQueryFromError(
+		return sdk.ABCIResponseQueryFromError(
 			std.ErrUnknownRequest(fmt.Sprintf(
 				"unknown vm query endpoint %s in %s",
 				secondPart(req.Path), req.Path)))
+	}
+
+	// Log the telemetry
+	logQueryTelemetry(path, res.IsErr())
```

Comment:
We could declare a constant for whether or not metrics are enabled and set it inside a file using build flags so that these checks are completely excluded from compilation when metrics are not enabled. The surround any call like this with `if metricsEnabled {`. The same applies to `keeper.go` as well. It might not make that much of a performance difference but it seems like we can just completely avoid it in this case. What do you think?